pkg/apis/internal.admin.acorn.io/v1/quotarequests.go

```diff
@@ -70,3 +73,50 @@ type QuotaRequestInstanceList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []QuotaRequestInstance `json:"items"`
 }
+type QuotaRequestResources struct {
+	BaseResources `json:",inline"`
+	Secrets       int `json:"secrets,omitempty"`
+}
+
+// Add will add the resources of another Resources struct into the current one.
+func (current *QuotaRequestResources) Add(incoming QuotaRequestResources) {
+	current.Secrets = Add(current.Secrets, incoming.Secrets)
+	current.BaseResources.Add(incoming.BaseResources)
+}
+
+// Remove will remove the resources of another Resources struct from the current one. Calling remove
+// will be a no-op for any resource values that are set to unlimited.
+func (current *QuotaRequestResources) Remove(incoming QuotaRequestResources, all bool) {
+	if all {
+		current.Secrets = Sub(current.Secrets, incoming.Secrets)
+	}
+	current.BaseResources.Remove(incoming.BaseResources, all)
+}
+
+// Fits will check if a group of resources will be able to contain
+// another group of resources. If the resources are not able to fit,
+// an aggregated error will be returned with all exceeded resources.
+// If the current resources defines unlimited, then it will always fit.
+func (current *QuotaRequestResources) Fits(incoming QuotaRequestResources) error {
+	exceededResources := []string{}
+	Fits(exceededResources, "Secrets", current.Secrets, incoming.Secrets)
```

Comment:
I noticed in `BaseResources`'s implementation of `Fits`, `exceededResources` was being set to the return value of `Fits` for each resource, like this:

```suggestion
	exceededResources := []string{}
	exceededResources = Fits(exceededResources, "Secrets", current.Secrets, incoming.Secrets)
```

Does that need to happen here as well?

Reply:
Good eye, yes.